reo/utilities.py

```diff
@@ -253,35 +253,103 @@ def check_common_outputs(Test, d_calculated, d_expected):
+    if not isinstance(consecutive_periods, list):
+        consecutive_periods = convert_dataframe_to_list_of_dict(consecutive_periods)
+
+    day_of_week_name = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
+    for i in range(len(consecutive_periods)):
+        start_month = int(consecutive_periods[i]["month"])  # One-indexed both user input and Calendar package
+        start_week_of_month = int(consecutive_periods[i]["start_week_of_month"] - 1)  # One-indexed for user, but zero-index for Calendar
+        start_day_of_week = int(consecutive_periods[i]["start_day_of_week"] - 1)  # Monday - Sunday is 1 - 7 for user, 0 - 6 for Calendar
+        start_hour = int(consecutive_periods[i]["start_hour"] - 1)  # One-indexed for user, datetime hour is zero-index
+        duration_hours = int(consecutive_periods[i]["duration_hours"])
+        error_start_text = "Error in chp_unavailability_period {}. ".format(i+1)
+        try:
+            start_day_of_month = calendar.Calendar().monthdayscalendar(year=year,month=start_month)[start_week_of_month][start_day_of_week]  # One-indexed
+            if start_day_of_month == 0:  # This may happen if there is no day_of_week in the 1st, 5th or 6th week of the month
+                raise DayOfWeekError("There is no start_day_of_week {} ({}) in week {} of month {} in the year {}. Remember, Monday is treated as the first day of the week.".format(start_day_of_week+1, day_of_week_name[start_day_of_week], start_week_of_month+1, start_month, year))
+            else:
+                start_datetime = datetime.datetime(year=year, month=start_month, day=start_day_of_month, hour=start_hour)
+                if start_datetime + datetime.timedelta(hours=duration_hours-1) > dt_profile[-1]:
+                    raise DurationOverflowsYearError("The start day/time and duration_hours exceeds the end of the year. Please specify two separate unavailability periods: one for the beginning of the year and one for up to the end of the year")
+                else:
+                    year_profile_hourly_series[start_datetime:start_datetime + datetime.timedelta(hours=duration_hours-1)] = 1.0
+            
+        except DayOfWeekError as e:
+            errors_list.append(error_start_text + str(e.args[0]))
+        except DurationOverflowsYearError as e:
+            errors_list.append(error_start_text + str(e.args[0]))
+        except:
+            errors_list.append(error_start_text + "Invalid set for month {} (1-12), start_week_of_month {} (1-4, possible 5 and 6), start_day_of_week {} (1-7), and start_hour_of_day {} (1-24) for the year {}.".format(start_month, start_week_of_month+1, start_day_of_week+1, start_hour+1, year))
+
+    if errors_list == []:
         year_profile_hourly_list = list(year_profile_hourly_series)
+    else:
+        year_profile_hourly_list = []   
+    
+    return year_profile_hourly_list, errors_list
```

Comment:
not a big deal at all, but just a thought. year_profile_hourly_list is always 0 or 1 right? You can avoid having to flip the numbers when setting the tech prod_factor if you just swap the logic in this function, i.e. make it an availability list. You'd need to adjust the views.py endpoint though (or let Andrew know to flip it too)

Reply:
Good thought, but I'm going to leave it as-is.